builtin/read-tree.c

```diff
@@ -223,11 +230,21 @@ int cmd_read_tree(int argc, const char **argv, const char *cmd_prefix)
 			opts.fn = opts.prefix ? bind_merge : oneway_merge;
 			break;
 		case 2:
+			/*
+			 * TODO: update twoway_merge to handle edit/edit conflicts in
+			 * sparse directories.
+			 */
+			ensure_full_index(&the_index);
 			opts.fn = twoway_merge;
 			opts.initial_checkout = is_cache_unborn();
 			break;
 		case 3:
 		default:
+			/*
+			 * TODO: update threeway_merge to handle edit/edit conflicts in
+			 * sparse directories.
+			 */
+			ensure_full_index(&the_index);
```

Comment:
My guess is that this case is rare enough that we might not even bother with it. Let's see if `git stash` ever needs this case, for instance. This `ensure_full_index()` might be a long-term solution.

Reply:
Since `git stash` only needs `oneway_diff` (single-tree) cases to work, I'll leave the `ensure_full_index` calls in for two-way and three-way merges.